core/storage/btree.rs

```diff
@@ -1755,138 +1755,169 @@ impl BTreeCursor {
+                        self.pager.load_page(page.clone())?;
+                        return Ok(CursorResult::IO);
+                    }
 
-            if let BTreeCell::TableLeafCell(leaf_cell) = cell {
-                if leaf_cell._rowid == target_rowid {
-                    cell_idx = Some(idx);
-                    break;
-                }
-            }
-        }
+                    target_rowid = match self.rowid.get() {
+                        Some(rowid) => rowid,
+                        None => return Ok(CursorResult::Ok(())),
+                    };
 
-        let cell_idx = match cell_idx {
-            Some(idx) => idx,
-            None => return Ok(CursorResult::Ok(())),
-        };
+                    let mut cell_idx = self.stack.current_cell_index() as usize;
+                    cell_idx -= 1;
+                    let contents = page.get().contents.as_ref().unwrap();
+                    let cell = contents.cell_get(
+                        cell_idx,
+                        self.pager.clone(),
+                        payload_overflow_threshold_max(
+                            contents.page_type(),
+                            self.usable_space() as u16,
+                        ),
+                        payload_overflow_threshold_min(
+                            contents.page_type(),
+                            self.usable_space() as u16,
+                        ),
+                        self.usable_space(),
+                    )?;
 
-        let contents = page.get().contents.as_ref().unwrap();
-        let cell = contents.cell_get(
-            cell_idx,
-            self.pager.clone(),
-            payload_overflow_threshold_max(contents.page_type(), self.usable_space() as u16),
-            payload_overflow_threshold_min(contents.page_type(), self.usable_space() as u16),
-            self.usable_space(),
-        )?;
+                    if cell_idx >= contents.cell_count() {
+                        return_corrupt!(format!(
+                            "Corrupted page: cell index {} is out of bounds for page with {} cells",
+                            cell_idx,
+                            contents.cell_count()
+                        ));
+                    }
 
-        if cell_idx >= contents.cell_count() {
-            return_corrupt!(format!(
-                "Corrupted page: cell index {} is out of bounds for page with {} cells",
-                cell_idx,
-                contents.cell_count()
-            ));
-        }
+                    let original_child_pointer = match &cell {
+                        BTreeCell::TableInteriorCell(interior) => Some(interior._left_child_page),
+                        _ => None,
+                    };
 
-        let original_child_pointer = match &cell {
-            BTreeCell::TableInteriorCell(interior) => Some(interior._left_child_page),
-            _ => None,
-        };
+                    return_if_io!(self.clear_overflow_pages(&cell));
```

Comment:
Is this re-entrant? I mean if we return I/O on any page inside `clear_overflow_pages` will we return to correct state, this feels like a case for a specific State for delete.